src/permissionSelectors.js

```diff
@@ -0,0 +1,37 @@
+/* @flow strict-local */
+import type { PerAccountState } from './types';
+// import { ensureUnreachable } from './types'; // (see commented callsite)
+import { Role, CreateWebPublicStreamPolicy } from './api/permissionsTypes';
+import { getRealm, getOwnUser } from './selectors';
+
+export function getCanCreateWebPublicStreams(state: PerAccountState): boolean {
+  const { webPublicStreamsEnabled, enableSpectatorAccess, createWebPublicStreamPolicy } = getRealm(
+    state,
+  );
+  const ownUser = getOwnUser(state);
+  const { role } = ownUser;
+
+  if (!webPublicStreamsEnabled || !enableSpectatorAccess) {
+    return false;
+  }
+
+  switch (createWebPublicStreamPolicy) {
+    case CreateWebPublicStreamPolicy.Nobody:
+      return false;
+    case CreateWebPublicStreamPolicy.OwnerOnly:
+      return role === Role.Owner;
+    case CreateWebPublicStreamPolicy.AdminOrAbove:
+      return role === Role.Owner || role === Role.Admin;
+    case CreateWebPublicStreamPolicy.ModeratorOrAbove:
+      return role === Role.Owner || role === Role.Admin || role === Role.Moderator;
+    default: {
+      // TODO: This `ensureUnreachable` should work. Uncomment it when it
+      //   does, or try Flow enums with the flowlint rule
+      //   require-explicit-enum-switch-cases, new in Flow 153.
+      // ensureUnreachable(createWebPublicStreamPolicy);
```

Comment:
Hmm, yeah, awkward.

I guess this is the same Flow issue as in 04c5a03d27a580182dce70c4b99260227614fb1d.

Though for that other instance, Flow enums aren't a solution, because there we're using the values as the "type" field of a disjoint object union, and [Flow enums don't support that](https://flow.org/en/docs/enums/using-enums/#toc-disjoint-object-unions) (because the values all have the same type, the enum type, instead of distinct types like string literals or number literals have.)

OTOH for that other instance, switching to literals was an OK workaround, because they're meaningful strings. Here, probably better to keep the names (as this branch does) and bite the bullet of having to carefully look for use sites whenever we add a new value to one of these enums.

Reply:
Same issue also as in https://github.com/zulip/zulip-mobile/pull/5382#discussion_r875160798 . I guess there the option I went for was:
```js
  switch (policy) {
    // FlowIssue: sad that we end up having to write numeric literals here :-/
    //   But the most important thing to get from the type-checker here is
    //   that the ensureUnreachable works -- that ensures that when we add a
    //   new possible value, we'll add a case for it here.  Couldn't find a
    //   cleaner way to write this that still accomplished that.
    case 0: // UserTopicVisibilityPolicy.None
      return false;
    case 1: // UserTopicVisibilityPolicy.Muted
      return true;
    default:
      ensureUnreachable(policy);
      throw new Error(`invalid UserTopicVisibilityPolicy: ${policy}`);
  }
```

Maybe that's better. There's still a bullet to bite: you have to hope, or check, that the numbers agree with the names in the comments. But it has the advantage that it's something you only have to redo when you're already looking at this code: unless we were to change the meaning of an existing value (which is much less likely than adding a new one), any faraway changes wouldn't silently invalidate this and would instead cause a type error here if it needs an update.

Reply:
Yeah, that makes sense—I'll do that, with your `FlowIssue` comment copied over, if you don't mind, and a link to this discussion.